File: frontend/src/app/modules/models/complaint.ts

export class Complaint {
  private _id: number;
  private _status: boolean;
  private _text: string;
  private _date: string;
  private _user_id: number;


  constructor(id: number, status: boolean, text: string, date: string, user_id: number) {
    this._id = id;
    this._status = status;
    this._text = text;
    this._date = date;
    this._user_id = user_id;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get status(): boolean {
    return this._status;
  }

  set status(value: boolean) {
    this._status = value;
  }

  get text(): string {
    return this._text;
  }

  set text(value: string) {
    this._text = value;
  }

  get date(): string {
    return this._date;
  }

  set date(value: string) {
    this._date = value;
  }

  get user_id(): number {
    return this._user_id;
  }

  set user_id(value: number) {
    this._user_id = value;
  }
}

File: fapi/src/main/java/com/netcracker/fapi/entity/Complaint.java
package com.netcracker.fapi.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;

public class Complaint {

    private long id;
    private boolean status;     /*checked/unchecked*/
    private String text;
    private String date;

    private User user;
    public User getUser() {
        return user;
    }
    public void setUser(User user) {
        this.user = user;
    }

    private Post post;
    public Post getPost() {
        return post;
    }
    public void setPost(Post post) {
        this.post = post;
    }




    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }

    public boolean isStatus() {
        return status;
    }
    public void setStatus(boolean status) {
        this.status = status;
    }

    public String getText() {
        return text;
    }
    public void setText(String text) {
        this.text = text;
    }

    public String getDate() {
        return date;
    }
    public void setDate(String date) {
        this.date = date;
    }
}
